### PR TITLE
Fix api corner cases and make chttpd pass

### PR DESCRIPTION
**Pull request by nickva**
### Overview

This is a set of commits to fix a few API corner cases and enable running chttpd as part of the `make check-fdb` suite.

### Checklist

```
  make check-fdb
```

### CODE REVIEW

File: src/chttpd/test/eunit/chttpd_open_revs_error_test.erl [hunk 1]
```diff
@@ -50,7 +50,7 @@ create_doc(Url, Id) ->
 delete_db(Url) ->
     {ok, 200, _, _} = test_request:delete(Url, [?AUTH]).
 
-open_revs_error_test_() ->
+open_revs_error_test_disabled() ->
```
Comment (davisp) [inline src/chttpd/test/eunit/chttpd_open_revs_error_test.erl:53]: Why not just delete these given that its a shards test?

File: src/chttpd/test/eunit/chttpd_security_tests.erl [hunk 2]
```diff
@@ -56,8 +55,10 @@ setup() ->
 teardown([Url,UsersUrl]) ->
     delete_db(Url),
     delete_db(UsersUrl),
+    ok = config:delete("chttpd_auth", "authentication_db", _Persist=false),
     ok = config:delete("admins", ?USER, _Persist=false).
 
+
```
Comment (davisp) [inline src/chttpd/test/eunit/chttpd_security_tests.erl:61]: whitespace only hunk

File: src/chttpd/test/eunit/chttpd_security_tests.erl [hunk 3]
```diff
@@ -236,14 +237,12 @@ should_disallow_anonymous_db_view_cleanup([Url,_UsersUrl]) ->
     ?_assertEqual(<<"unauthorized">>, ErrType).
 
 should_allow_admin_purge([Url,_UsersUrl]) ->
```
Comment (davisp) [inline src/chttpd/test/eunit/chttpd_security_tests.erl:239]: Should we do the `_disabled` thing? Alternatively, does eunit have a notion of skipped tests?

Comment (davisp) [reply src/chttpd/test/eunit/chttpd_security_tests.erl:239]: Gonna say samesies for all tests switched to 501 assertions since we'll need to undo them once those features are actually implemented.

Comment (nickva) [reply src/chttpd/test/eunit/chttpd_security_tests.erl:239]: Was on the fence about this one since it's not really testing _purge functionality but testing security so I thought we'd keep letting the tests run in the meantime before we implement it. But I don't think we're gaining a lot more coverage here anyway so will disable them.

File: src/chttpd/test/eunit/chttpd_security_tests.erl (hunk 2 above)
Comment (nickva) [reply src/chttpd/test/eunit/chttpd_security_tests.erl:61]: doh, will clean it up

File: src/chttpd/test/eunit/chttpd_open_revs_error_test.erl (hunk 1 above)
Comment (nickva) [reply src/chttpd/test/eunit/chttpd_open_revs_error_test.erl:53]: Good call will delete it. I started with wanting to minimize the diff sizes between master but deleted vs not is not that big of a deal I think

File: src/chttpd/src/chttpd_db.erl [hunk 4]
```diff
@@ -252,27 +252,19 @@ maybe_flush_changes_feed(Acc0, Data, Len) ->
...
-            end
-    end;
+    send_json(Req, 202, {[{ok, true}]});
 
 handle_compact_req(Req, _Db) ->
     send_method_not_allowed(Req, "POST").
 
-handle_view_cleanup_req(Req, Db) ->
-    ok = fabric:cleanup_index_files_all_nodes(Db),
+handle_view_cleanup_req(Req, _Db) ->
```
[the remark this answers is not in the document]
Comment (nickva) [reply src/chttpd/src/chttpd_db.erl:267]: Good idea.I should have remembered it. I just copied the compactor behavior without thinking. Will make it a 501

File: src/chttpd/test/eunit/chttpd_security_tests.erl (hunk 3 above)
Comment (nickva) [reply src/chttpd/test/eunit/chttpd_security_tests.erl:239]: I think the easiest thing would be to comment out the tests in the eunit test definition block at the top. There is no skip / or disable switch in eunit that I could find.

**Comment by nickva** [comment]
@davisp thank you for taking a look!

Fixed up most of the things as requested.

**Comment by nickva** [comment]
@jaydoane I added compiler directives to ignore unused functions in those two module and a comment for it so we know when to remove it in the future
